### DISCUSSION TITLE
Creat a trait

### DISCUSSION THREAD

**Discussion by JuniorVezintana** [GrapesJS/grapesjs#5526] [Q&A]
Any ideas on how to create a trait for the 'details' tag that applies open to the tag, so that it is open by default or closed if the checkbox is already unchecked?

**Comment by artf** [comment]
This is the default behavior of the trait checkbox type
```js
attributes: { open: true }, // e.g make it checked by default
traits: [
  {
    type: 'checkbox',
    name: 'open',
  },
]
```

**Reply by JuniorVezintana** [reply]
For me it's not standard, in any element or block, only the id and title come, I can't add any other trait to any element!
Maybe I'm adding it wrong. Are traits defined in the block or at initialization?
////////////////////////////////
$this.objBlocksDefault['section-ini'] = {
            label: 'Seção',
            category: $this.strBlockText,
            media: `<img src="https://cdn.plices.com/assets/app/builder/svg/column-1.jpg" style="width:100%">`,
            attributes: { class: '', title: "Adicione uma seção ao layout" },
            editable: true,
            draggable: true,
            selectable: true,
            resizable: true,
            trait: {
                type: 'color',
                // ...
              },
            content: `
                    <section style="width: 100%; font-family: inherit; min-height: 150px; display: flex; flex-direction: column; justify-content: center; align-items: center; box-sizing: border-box; padding: 10px;" data-gjs-resizable="true"></section>
                    `
        };

**Reply by artf** [reply]
> Maybe I'm adding it wrong

yes... you're confusing [component properties](https://grapesjs.com/docs/api/component.html#properties) with [block ones](https://grapesjs.com/docs/api/block.html#properties), please read carefully the documentation.
